plugins/scripting/scripting-ide-services/src/org/jetbrains/kotlin/scripting/ide_services/compiler/impl/KJvmReplCompleter.kt

```diff
@@ -238,22 +241,34 @@ private class KJvmReplCompleter(
                 .let {
                     if (isSortNeeded) it.sortedBy { descTriple -> descTriple.third } else it
                 }
-                .forEach {
-                    val descriptor = it.first
-                    val (rawName, presentableText, tailText, completionText) = it.second
+                .forEach { resultTriple ->
+                    val descriptor = resultTriple.first
+                    val (rawName, presentableText, tailText, completionText) = resultTriple.second
                     if (nameFilter(rawName, prefix)) {
                         val fullName: String =
                             formatName(
                                 presentableText
                             )
+                        val deprecationLevel = descriptor.annotations
+                            .findAnnotation(FqName("kotlin.Deprecated"))
+                            ?.let {
+                                val valuePair = it.argumentValue("level")?.value as? Pair<*, *>
+                                val valueName = (valuePair?.second as? Name)?.identifier ?: return@let DeprecationLevel.WARNING
+                                try {
+                                    DeprecationLevel.valueOf(valueName)
+                                } catch (e: IllegalArgumentException) {
```

Comment:
Please add a comment, why such a catch is needed here. It looks hacky to me.

Reply:
Changed the logic here a bit, removed try/catch